### PR TITLE
fixes radiation randomly killing air

### PR DESCRIPTION
**Pull request by Bod9001**
CL: [Fix] Made radiation work across matrices
CL: [Fix] Made explosions Work across matrices

### CODE REVIEW

**Review by NoooneyDude — DISMISSED** [review]

**Comment by Bod9001** [comment]
note :  matrix get point needs to be more thread safe

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 1]
```diff
@@ -770,6 +770,11 @@ private List<Color> GetPaletteOrNull()
 	public void UpdateMe()
 	{
 		timeElapsed += UpdateManager.CashedDeltaTime;
+		if (PresentFrame == null)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:773]: is this also related to radiation killing atmos?

File: UnityProject/Assets/Scripts/HealthV2/Living/CirculatorySystem/BodyPartBlood.cs [hunk 2]
```diff
@@ -82,6 +82,14 @@ public partial class BodyPart
 		/// </summary>
 		public float BloodThroughput => bloodThroughput;
 
+		public enum RequiredBlood
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/CirculatorySystem/BodyPartBlood.cs:85]: nani

Comment (corp-0) [reply UnityProject/Assets/Scripts/HealthV2/Living/CirculatorySystem/BodyPartBlood.cs:85]: also prefer to use the HealthConstants file

File: UnityProject/Assets/Scripts/Player/PlayerScript.cs [hunk 3]
```diff
@@ -345,7 +345,7 @@ public void SyncPlayerName(string oldValue, string value)
 	/// <summary>
 	/// True if this player is a ghost, meaning they exist in the ghost layer
 	/// </summary>
-	public bool IsGhost => PlayerUtils.IsGhost(gameObject);
+	public bool IsGhost => PlayerState == PlayerStates.Ghost;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/PlayerScript.cs]: This might break Ai and blob

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:773]: nah, I just got annoyed with it being spammed on client

File: UnityProject/Assets/Scripts/HealthV2/Living/CirculatorySystem/BodyPartBlood.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/CirculatorySystem/BodyPartBlood.cs:85]: yeah, was redoing bloodflow stuff, but then changed over to radiation

File: UnityProject/Assets/Scripts/Player/PlayerScript.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Player/PlayerScript.cs]: are they checked as is ghost?

I can add the Ai  and  blob if you want

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Player/PlayerScript.cs]: The IsGhost is used in some places which the AI and blob need to be checked for as well, i never got around to seeing which stuff really needed it so they are just ghosts atm.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Player/PlayerScript.cs]: ok will add them

File: UnityProject/Assets/Scripts/Items/Tool/Syringe.cs [hunk 4]
```diff
@@ -26,6 +26,9 @@ public void ServerPerformInteraction(HandApply interaction)
 		if (LHB != null)
 		{
 			LHB.CirculatorySystem.ReadyBloodPool.Add(LocalContainer.TakeReagents(10f));
+			Chat.AddActionMsgToChat(interaction.Performer, "You Inject The syringe in to " + LHB.playerScript.name,
+				interaction.Performer.name + " injects a syringe in to " + LHB.playerScript.name);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Tool/Syringe.cs]:
```suggestion
			Chat.AddActionMsgToChat(interaction.Performer, $"You Inject The syringe into {LHB.gameObject.ExpensiveName()}",
				$"{interaction.Performer.ExpensiveName()} injects a syringe into {LHB.gameObject.ExpensiveName()}");
```
LHB.playerScript.visibleName might also work too

**Review by NoooneyDude — APPROVED** [review]

**Review by ThatDan123 — APPROVED** [review]
